docs/en/observability/rate-aggregations.asciidoc

```diff
@@ -0,0 +1,78 @@
+[[understanding-rate-aggregations]]
+= Understanding rate aggregations
+
+You can use a rate aggregation to analyze the rate at which a specific field changes over time.
+This type of aggregation is useful when you want to analyze fields like counters.
+
+For example, imagine you have a counter field called restarts that increments each time a service restarts.
+You can use rate aggregation to get an alert if the service restarts more than X times within a specific time window (for example, per day).
+
+[discrete]
+[[how-rates-are-calculated]]
+== How rates are calculated
+
+Rates used in alerting rules are calculated by comparing the maximum value of the field in the previous bucket to the maximum value of the field in the current bucket and then dividing the result by the number of seconds in the selected interval.
+For example, if the value of the restarts increases, the rate would be calculated as:
+
+`(max_value_in_current_bucket - max_value_in_previous_bucket)/interval_in_seconds`
+
+In this example, let’s assume you have one document per bucket with the following data:
+
+[source,json]
+----
+{
+"timestamp": 0000,
+"restarts": 0
+}
+
+{
+"timestamp": 60000,
+"restarts": 1
+}
+
+----
+
+// I might be a little timestamp naive, but I don't understand why we start with 0000 in this example (or whether it matters either way).
+
+Let’s assume the timestamp is a UNIX timestamp in milliseconds,
+and we started counting on Thursday, January 1, 1970 12:00:00 AM.
+In that case, the rate will be calculated as follows:
+
+`(max_value_in_current_bucket - max_value_in_previous_bucket)/interval_in_seconds`, where:
+
+* `max_value_in_current_bucket` (now-1m → now): 1
+* `max_value_in_previous_bucket` (now-2m → now-1m): 0
+* `interval_in_seconds`: 60
+
+The rate calculation would be: `(1 - 0) / 60 = 0.0166666666667`
+
+If you want to alert when the rate of restarts is above 1 within a 1-minute window, you would set the threshold above `0.0166666666667`.
+
+// I think it might be confusing that we are representing the calculation in two different ways here,
+// especially since the first row in the table shows the calculation from the example.
+// At the very least, I think we need some kind of transition that introduces this table.
+// Maybe something like this (please feel free to suggest something better):
+
+The calculation you need to use depends on the interval that's selected:
+
+|===
+| Interval | Calculation
+
+| 1m
+| restarts from [now-1m → now] - restarts from [now-2m → now-1m] / 60
```

Comment:
I see above I used `(now-1m → now)` and here `[now-1m → now]`, maybe we can use one of the styles. Do we have such representation in other documents for an interval?

Reply:
Hmm...good question. If this convention exists somewhere else in the docs, I'm not aware of it.

Reply:
I'll use square brackets. We will be doing a lot of cleanup during the asciidoc > docsmobile migration, so I'm not too concerned about this small potential inconsistency right now.